plugins/ui/edgy.nix

```diff
@@ -0,0 +1,321 @@
+    in
+    {
+      left = mkViewOptsOption "left";
+      bottom = mkViewOptsOption "bottom";
+      right = mkViewOptsOption "right";
+      top = mkViewOptsOption "top";
+
+      options =
+        mapAttrs
+          (_: defaultSize: {
+            size = helpers.defaultNullOpts.mkUnsignedInt defaultSize ''
+              Size of the short edge of the edgebar.
+              For edgebars, this is the minimum width.
+              For panels, minimum height.
+            '';
+
+            wo = helpers.mkNullOrOption (with types; attrsOf anything) ''
+              View-specific window options.
+            '';
+          })
+          {
+            left = 30;
+            bottom = 10;
+            right = 30;
+            top = 10;
+          };
+
+      animate = {
+        enabled = helpers.defaultNullOpts.mkBool true ''
+          Whether to enable animations.
+        '';
+
+        fps = helpers.defaultNullOpts.mkUnsignedInt 100 ''
+          Frames per second.
+        '';
+
+        cps = helpers.defaultNullOpts.mkUnsignedInt 120 ''
+          Cells per second.
+        '';
+
+        on_begin = helpers.defaultNullOpts.mkLuaFn ''
+          function()
+            vim.g.minianimate_disable = true
+          end
+        '' "Callback for the beginning of animations.";
+
+        on_end = helpers.defaultNullOpts.mkLuaFn ''
+          function()
+            vim.g.minianimate_disable = false
+          end
+        '' "Callback for the ending of animations.";
+
+        spinner = mkOption {
+          type =
+            with helpers.nixvimTypes;
+            nullOr (
+              either strLua (submodule {
+                freeformType = attrsOf anything;
+                options = {
+                  frames = helpers.defaultNullOpts.mkListOf types.str [
+                    "⠋"
+                    "⠙"
+                    "⠹"
+                    "⠸"
+                    "⠼"
+                    "⠴"
+                    "⠦"
+                    "⠧"
+                    "⠇"
+                    "⠏"
+                  ] "Frame characters.";
+
+                  interval = helpers.defaultNullOpts.mkUnsignedInt 80 ''
+                    Interval time between two consecutive frames.
+                  '';
+                };
+              })
+            );
+          default = null;
+          example = "require('noice.util.spinners').spinners.circleFull";
+          apply = v: if isString v then helpers.mkRaw v else v;
+          description = ''
+            Spinner for pinned views that are loading.
+
+            ${helpers.defaultNullOpts.mkDefaultDesc {
+              frames = [
+                "⠋"
+                "⠙"
+                "⠹"
+                "⠸"
+                "⠼"
+                "⠴"
+                "⠦"
+                "⠧"
+                "⠇"
+                "⠏"
+              ];
+              interval = 80;
+            }}
+          '';
+        };
+      };
```

Comment:
This would benefit from some explanatory comments. I'd also like @traxys' thoughts on this before merging, since I'm struggling to follow the what & why here.